### ISSUE TITLE
Make community

### ISSUE THREAD

**Issue by MariadeAnton** [biicode/biicode#12]
Gamification:

  - Keep track of interesting blocks
  - Get notifications
  - Know when someone "reuses" your code
  - An easy way to contact block’s owners

**Comment by Manu343726** [comment]
What's the state of the chat on blocks feature?

**Comment by MariadeAnton** [comment]
A user suggested we could have user-email accounts in which you could contact (via biicode) a user similar to github's email accounts.

Otherwise the disqus chat on the bottom of each block's page seems like a nice feature.

**Comment by MariadeAnton** [comment]
Also @metamaker says: It would be cool to have some kind of issue tracker for blocks at www.biicode.com or any other way to contact block's owner directly (IM, PM, emailing, etc). 

http://forum.biicode.com/t/feature-request-a-way-to-contact-blocks-owner-at-biicode-com/409

**Comment by metamaker** [comment]
Hello!

Since PM functionality is already present at forum, merging biicode.com and forum.biicode.com user accounts could nicely solve the problem. I believe it is also possible to enable emailing between users in Discourse (through web form like in phpBB).

Chat on blocks is quite nice feature but issue tracker is better since you can organize messages in a way you want, mark them with tags, etc (you know, that cool stuff on git).

For projects located at github it would be nice to provide a link to github repository on top of block's page (though the link could be just mentioned in readme.md but I haven't see many projects providing links back to github). And again such feature could be implemented for other sites like sourceforge, travis-ci, project's website, etc to wire all things together.

**Comment by drodri** [comment]
Hi @metamaker! Actually, there is such a link, check: https://www.biicode.com/lasote/lasote/libuv/v0.10

The link to github is created using option "-r" = remote in bii publish:

```
$ bii publish -r
```
If currently in a github repo, it will obtain the info, and it will query the user for an url otherwise.
Maybe it should be more visible/highlighted.

About the forum, that is right, we have to merge (OAuth accounts for forum and biicode), the issue tracker should be the one of the VCS that supports the code (github, etc). It would be great if we could provide some integration via API, or whatever, but that will take some time. Thanks!

**Comment by metamaker** [comment]
> Maybe it should be more visible/highlighted.

Yep, I don't see this `-r` option at either <http://docs.biicode.com/c++/publishing.html> or <http://docs.biicode.com/c++/reference/commands/publish.html> though it is described at <http://docs.biicode.com/c++/integration/git.html#publish-from-git-commit>. Linking publishing guide to git integration page and mentioning of `-r` flag in reference could be wise.

> the issue tracker should be the one of the VCS that supports the code (github, etc)

I agree, that is good idea not to create new bycicle but rather use existing solution. Does that mean that bii is not intended to provide VCS functions itself (even in future) but rather be used only as software for publishing libraries releases?

> It would be great if we could provide some integration via API, or whatever, but that will take some time.

It is possible for **github**, I don't know if for others too.

**Comment by Manu343726** [comment]
> Does that mean that bii is not intended to provide VCS functions itself (even in future) but rather be used only as software for publishing libraries releases?

This is exactly what bii is supposed to be. Projects are hosted and VCSed in existing and well suited services such as github, bitbucket, etc; then released and shipped to bii. biicode provides dependency management and user-ready library hosting. We are planning to support [external storage](https://github.com/biicode/biicode/issues/6) too, in a way you can still host your lib in its github repo for example, and bii only stores dependency management info.
